app/models/manageiq/providers/hawkular/datawarehouse_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,105 @@
+class ManageIQ::Providers::Hawkular::DatawarehouseManager::EventCatcher::Runner <
```

Comment:
The intention here was to write the code so that it could be used by both consumers and extract the middleware code in a different PR (by me or by someone from middleware that can better test the change). If we try to do both here I fear this would not get merged.

(cc @Jiri-Kremser @lucasponce @simon3z)